crates/astria-sequencer-relayer/src/relayer/write/mod.rs

```diff
@@ -297,26 +366,22 @@ async fn submit_blobs(
     let largest_sequencer_height = data.greatest_sequencer_height();
     let blobs = data.into_blobs();
 
-    let submission_started = match crate::utils::flatten(
-        tokio::task::spawn_blocking(move || submission_state.initialize(largest_sequencer_height))
-            .in_current_span()
-            .await,
-    ) {
-        Err(error) => {
-            error!(%error, "failed to initialize submission; abandoning");
-            return Err(error);
-        }
-        Ok(state) => state,
-    };
+    let new_state = submit_with_retry(
+        client,
+        blobs,
+        state.clone(),
+        started_submission,
+        largest_sequencer_height,
+        metrics,
+    )
+    .await
+    .map_err(|error| {
+        let message = "failed submitting blobs to Celestia";
+        error!(%error, message);
```

Comment:
IMO don't emit an `error!` here but change the `instrument` macro to read `#[instrument(skip_all, err)]`

Reply:
Nice - I didn't realise that would have the same effect as explicitly logging like this.

Done in [9ade24c](https://github.com/astriaorg/astria/pull/1234/commits/9ade24cf8a1b470ef1ba8524afad2fcbab2dd4ab).